analyzer/queries.go

```diff
@@ -429,14 +429,26 @@ func (qf QueryFactory) AddressPreimageInsertQuery() string {
 		ON CONFLICT DO NOTHING`, qf.chainID)
 }
 
-func (qf QueryFactory) RuntimeTokenChangeUpdateQuery() string {
+func (qf QueryFactory) RuntimeTokenBalanceUpdateQuery() string {
 	return fmt.Sprintf(`
 		INSERT INTO %[1]s.%[2]s_token_balances (token_address, account_address, balance)
 			VALUES ($1, $2, $3)
 		ON CONFLICT (token_address, account_address) DO
 			UPDATE SET balance = %[1]s.%[2]s_token_balances.balance + $3`, qf.chainID, qf.runtime)
 }
 
+func (qf QueryFactory) RuntimeTokenUpdateQuery() string {
+	return fmt.Sprintf(`
+		INSERT INTO %s.%s_tokens (token_address, token_name, symbol, decimals, total_supply)
```

Comment:
that we should have combined tables that included data from multiple paratimes (right?)--I'm in favor

Reply:
On one hand, combining by SDK module makes sense. Adding new ParaTimes would just be a matter of right configuration.

On the other, grouping them by "analyzer instances" (I suppose this is the current situation) also makes sense. With this approach by looking at a table you know which "analyzer instance" generated it. If an "analyzer instance" starts having issues, you also know exactly which table is completely behind the latest round. In case of data corruption or the need to rollback, it is also much easier to rollback just one table or drop the table and re-run the "analyzer instance" again, instead of rolling back or dropping just a part of the table (huh!?).

Reply:
would doing this interfere with grouping by "analyzer instances"? how do we do our databases for multiple analyzer instances currently?

Reply:
> On one hand, combining by SDK module makes sense. Adding new ParaTimes would just be a matter of right configuration.
This is a strong motivating factor IMO, especially given the similarities between Emerald and Sapphire (and the desire to launch Sapphire pretty fast).

 For your other points, @gw0: True, but mostly negated by the fact that an "analyzer instance" (these are per-paratime, if you count consensus as a paratime; you could have multiple instances per paratime, but then the current db design doesn't separate their data either) will always write the paratime name into a column of all these paratime-shared tables. So "dropping" becomes a little more complicated ("DELETE FROM ... WHERE paratime=...") but still doable in a pinch. And for local testing or total from-scratch redeploys (= most common use cases, I expect), it's still OK to go nuclear and wipe all the analyzers' state.

Reply:
Oh, if the current db design does not separate paratime data, then I guess it is time to change... :)

After sleeping over it, I believe combining by SDK module and storing the paratime name into a column is actually much better. Potential bugs that need dropping and re-indexing are probably contained within an SDK module, and re-running just one such analyzer for a paratime should be much faster than all of them. Thanks for clarifying!

Reply:
In the `operators-health` project table names are a configurable parameter for each "runner" (the equivalent of analyzer). Not sure if it is the best way, but it is definitely the most flexible way.

Reply:
:+1: separate PR is fine

Reply:
filed 262, thanks for bringing this up. closing for merge